src/diagnostic-channel-publishers/src/pg.pub.ts

```diff
@@ -0,0 +1,160 @@
+                if (err) {
+                    if (cb) {
+                        return cb.apply(this, arguments);
+                    } else if (queryResult && queryResult instanceof EventEmitter) {
+                        queryResult.emit("error", err);
+                    }
+                } else if (cb) {
+                    cb.apply(this, arguments);
+                }
+            });
+
+            try {
+                Object.defineProperty(trackingCallback, diagnosticOriginalFunc, { value: cb });
+                return trackingCallback;
+            } catch (e) {
+                // this should never happen, but bailout in case it does
+                return cb;
+            }
+        }
+
+        // this function takes too many variations of arguments.
+        // this patches any provided callback or creates a new callback if one wasn't provided.
+        // since the callback is always called (if provided) in addition to always having a Promisified
+        // EventEmitter returned (well, sometimes -- see above), its safe to insert a callback if none was given
+        try {
+            if (typeof config === "string") {
+                if (values instanceof Array) {
+                    data.query.preparable = {
+                        text: config,
+                        args: values,
+                    };
+                    callback = patchCallback(callback);
+                } else {
+                    data.query.text = config;
+
+                    // pg v6 will, for some reason, accept both
+                    // client.query("...", undefined, () => {...})
+                    // **and**
+                    // client.query("...", () => {...});
+                    // Internally, precedence is given to the callback argument
+                    if (callback) {
+                        callback = patchCallback(callback);
+                    } else {
+                        values = patchCallback(values);
+                    }
+                }
+            } else {
+                if (typeof config.name === "string") {
+                    data.query.plan = config.name;
+                } else if (config.values instanceof Array) {
+                    data.query.preparable = {
+                        text: config.text,
+                        args: config.values,
+                    };
+                } else {
+                    data.query.text = config.text;
+                }
+
+                if (callback) {
+                    callback = patchCallback(callback);
+                } else if (values) {
+                    values = patchCallback(values);
+                } else {
+                    config.callback = patchCallback(config.callback);
+                }
+            }
+        } catch (e) {
+            // if our logic here throws, bail out and just let pg do its thing
+            return originalClientQuery.apply(this, arguments);
```

Comment:
Id guess that would be on a new channel?

```ts
export interface IPatchError {
  module: string,
  argConfig: string[] // typeof each arg
}
channel.publish("patch-error", { module: "pg", argConfig: Array.prototype.slice.call(arguments).map(arg => typeof arg) });
```

We wouldn't want to include actual args since those might contain sensitive info, and this would require a bit more work with objects to inspect them more.